### ISSUE TITLE
[Rebuild] proton-ge-custom

### ISSUE THREAD

**Issue by guiltiest-gear** [chaotic-aur/packages#2870]
### Packages:

https://aur.archlinux.org/packages/proton-ge-custom

### Latest chaotic version

2:GE.Proton8.20-1

### Reason for rebuild

2:GE.Proton8.21-1 is current version in AUR

### Logs

https://builds.garudalinux.org/repos/chaotic-aur/logs/proton-ge-custom.log

**Comment by xiota** [comment]
In the future, please comment on the old rebuild requests to prevent proliferation of duplicate issues.

Also, please wait a day and resync your package database before requesting rebuilds.  The build server has already automatically rebuilt this package.

**Comment by niobium93** [comment]
@xiota Needs to be rebuilt again. `GE.Proton9.9-1` is the latest in the AUR. [Log](https://builds.garudalinux.org/repos/chaotic-aur/logs/proton-ge-custom.log) seems to show some issue:
```
error: libxxf86vm: could not find package in cache
```

**Comment by xiota** [comment]
Unfortunately, the builders are down, so nothing can be rebuilt.  Something related to that error is blocking everything.

A new build system is being prepared.  There are still hundreds of missing packages and other issues.  However, if you're willing to test, you can switch by changing the config in `/etc/pacman.conf`:
```
[chaotic-aur]
Server = https://builds.garudalinux.org/chaotic-v4/$arch
```

**Comment by dr460nf1r3** [comment]
Wiped the package cache once again.. hopefully, it stays this time.

**Comment by niobium93** [comment]
Why is this package seemingly getting rebuilt over and over again? We are up to version `2:GE.Proton9.20-1.4` now with seemingly 4 rebuilds of the same package. Every day is another seemingly identical ~400MB download.

**Comment by dr460nf1r3** [comment]
There were unnecessary rebuilds due to a bug in the bump system a few days ago. If it keeps persisting, please notify us here. In general, the bumps should only be happening if shared library requires it. But the current implementation (after the big fix) will inevitably end of up causing a few useless rebuilds. We are currently evaluating whether it's worth taking the risk.

**Comment by niobium93** [comment]
@dr460nf1r3 `proton-ge-custom` has now rebuilt for the 5th time. Everyday it installs another one of these. I assume it's getting rebuild every time one of it's many dependencies gets updated? Reinstalling a 400mb package everyday feels like pointless churn for my ssd. `cpu-x` is another package that has gotten 5 pointless rebuilds by now.

**Comment by dr460nf1r3** [comment]
Yes, sure. I've set a cooldown period to 7 days. Will keep investigating how to prevent false positives.

**Comment by yan12125** [comment]
wine-stable is also affected by false positives (#311)

@dr460nf1r3 Mind to share some technical details of the issue?

**Comment by dr460nf1r3** [comment]
> wine-stable is also affected by false positives (#311)
> 
> @dr460nf1r3 Mind to share some technical details of the issue?

The situation we have now originates in trying to set up a semi-intelligent bump system. So far we had to rebuild things manually, which basically requires users to report the lib mismatches. There are a few ways to execute a bump right now:

- explicitly (adding dep to pkgbase config, triggering a rebuild whenever said package pkgver changes)
- globally defined triggers (the same, but globally)
- Comparing archlinux pkgbase provides/namcap derived .so providers against a list of .so deps of packages (also derived from namcap analysis) 

If any of those match, a rebuild would be executed by pushing a bump commit.

It was clear especially the last one would cause at least a few unnecessary rebuilds. It is certainly not flawless, but I'm currently unsure how to reduce the size of false matches. If you have an idea, please let us hear it 😸

**Comment by yan12125** [comment]
> I guess that would be an idea!

@dr460nf1r3 Is it going to happen?

If you're too busy on this, I'm willing to contribute it. However, I cannot find relevant scripts in this github organization: https://github.com/search?q=org%3Achaotic-aur+Namcap+analysis&type=code

Unnecessary rebuilds have been an annoying issue. Besides wine-stable, insync is also affected (https://github.com/search?q=org%3Achaotic-aur+insync&type=commits&s=committer-date&o=desc). It's a proprietary application, so it's likely rebuilds are unnecessary.

**Comment by xiota** [comment]
Is there a bump exception list?

**Comment by yan12125** [comment]
I'm not sure - apparently the scripts for bumping are not public, so I guess the exception list is not, either. Also, an exception list may not be a good solution. Apparently there are quite a few broken cases. Such a list may be a headache to maintain. Please consider identifying the root cause and fix it.

**Comment by dr460nf1r3** [comment]
> If you're too busy on this, I'm willing to contribute it

That would be very kind of you, and is appreciated. I was quite busy with creating the new website lately. 

The reason for not finding it in the repository itself is that it is happening in the backend, specifically because we have access to much more information than in the CI pipelines via sets of data saved to our database. The relevant stuff can be found here: 

https://github.com/chaotic-cx/chaotic-next/blob/main/backend/src/repo-manager/repo-manager.service.ts#L759

The namcap logic is here: 

https://github.com/chaotic-cx/chaotic-next/blob/main/backend/src/repo-manager/repo-manager.service.ts#L1536

I'm assuming that maybe changing the keys checked would result in less inaccurate bumps.

The new website also displays the detected .so dependencies, so you'll be easily able to know what exactly will get triggered. Looking at it, the list of .so deps is indeed HUGE 🤯  -> https://aur.chaotic.cx/stats?search=proton-ge-custom

**Comment by yan12125** [comment]
Thank you for the pointers. It seems that a package is bumped if namcap detects .so names from another just-built package. Generally, the so version should also be checked. For example, bumping only when libfoo.so.2 becomes libfoo.so.3, not whenever package foo is updated.

There are indeed some edge cases with ABI changes but no soversion changes. Those cases can be handled exceptionally by bypassing the soversion check.

**Comment by xiota** [comment]
If problem cannot be resolved by adjusting bump logic, I can probably override them in interfere.  Will wait a while before attempting.

**Comment by dr460nf1r3** [comment]
This is definitely an issue we can and *need* to fix. I agree with this. I'll attempt doing so tomorrow. Sorry for getting back to this late, I had a very busy week. :)
